mailbox/api/pom.xml

```diff
@@ -45,10 +49,6 @@
             <artifactId>james-server-task-memory</artifactId>
             <scope>test</scope>
         </dependency>
-        <dependency>
-            <groupId>${james.groupId}</groupId>
-            <artifactId>apache-mime4j-dom</artifactId>
```

Comment:
this one is trickier because it's not really an internal POM. Maybe we should replace ${james.groupId} by org.apache.james here?

Reply:
I don't understand : how can it not be internal if it is under our groupId ?

Reply:
mime4j is a project of org.apache.james. But it's not in the same repository and has not the same versioning than James.
That's why I think here we should explicit "org.apache.james".

Reply:
I will make the change.

However I would like to point out https://maven.apache.org/guides/mini/guide-naming-conventions.html
> groupId uniquely identifies your project across all projects. [...]  That is, if the current project is a multiple module project, it should append a new identifier to the parent's groupId.

Reply:
I guess it's historical. Probably we should rename mime4j's group id now, but it's not a trivial change :)

Reply:
I have added a commit which explicitely drops the `${james.groupId}` for all the `apache-mime4j-xxx` artifacts

Reply:
>  Probably we should rename mime4j's group id now

:+1: